Add search to members and staff tables in organization page

vgeorge opened the pull request:
This adds an input component to the org members table to allow search by username. Contributes to https://github.com/developmentseed/osm-teams/issues/319. 

I've moved username caching to `Team.addMember`. This was necessary to ensure all users have usernames initialized. We will have to use this approach for all searchable routes. cc @kamicut

vgeorge commented:
I've also added search to the org staff table. The search input needs styling.

@LanesGood @kamicut this is ready for review.

LanesGood commented:
@vgeorge this is working well. I've added a few style changes.

A few questions:
- how can a user get back to the original table? I assumed that deleting my query from the search box and then clicking/focusing out of the box would get back to the original table, or deleting the query and hitting enter, but these approaches don't reset the table. Instead, I have to delete my query and click "search" again. 
- Would it be possible to add a "Clear" button to the search input? It looks like setting the html input type to `type=search` should accomplish this, but I'm not getting it to work
- If a search returns no results I see "No members yet." Could this be set to a message like "search returned no results" instead?

![image](https://user-images.githubusercontent.com/12634024/211671353-f479d617-18a3-4c9d-a028-91bdd0a91aa0.png)

vgeorge commented:
@LanesGood Thanks for the review. I added auto-submit to the input. If the user clears the field the table will be reset to the original state. On Chrome I could see the 'X' icon for clearing the input, but not on Firefox. Maybe the clear button is not necessary with the auto-submit? I also updated the "empty table message" to display 'Search returned no results.' when a search term is present. This is ready for another review.

LanesGood reviewed (approved):
@vgeorge thanks for introducing these changes. I've made some changes for the button and search input styles. These are a bit verbose, but as we'll plan to replace them with Chakra/a new design I didn't think it was worth refining (and I keep running into issues when trying to render styles conditionally with Styled JSX).

kamicut reviewed (approved):
Looks good to me. I think we can resolve usernames in a more concise way if you'd like to follow my recommendations, but I don't want to block the other PRs.

vgeorge commented on src/models/team.js:
```diff
@@ -344,6 +344,9 @@ async function create(data, osmId, trx) {
     })
   }
 
+  // Cache username
```
@kamicut I believe the only way to implement a catch-all is to add a trigger on user DB inserts that fetches the username from OSM, but I don't see a clear way of implementing this in knex.js. Also, we have to apply this trigger to all tables where the user can exist solely ('member', 'organization_manager', 'users'), which doesn't look cleaner than the implementation added in this PR. I'm going to merge this now, let's open a ticket if we consider it worth looking for a refactor.
